index.js

```diff
@@ -2,70 +2,64 @@
 
 const eu = encodeURIComponent
 const fetch = require('npm-registry-fetch')
-const figgyPudding = require('figgy-pudding')
-const getStream = require('get-stream')
 const validate = require('aproba')
 
-const OrgConfig = figgyPudding({
-  Promise: { default: () => Promise }
-})
-
 // From https://github.com/npm/registry/blob/master/docs/orgs/memberships.md
 const cmd = module.exports = {}
 
 class MembershipDetail {}
-cmd.set = (org, user, role, opts) => {
-  if (typeof role === 'object' && !opts) {
+cmd.set = (org, user, role, opts = {}) => {
+  if (typeof role === 'object' &&
+  Object.keys(opts).length === 0) {
     opts = role
     role = undefined
   }
-  opts = OrgConfig(opts)
-  return new opts.Promise((resolve, reject) => {
+  return new Promise((resolve, reject) => {
```

Comment:
You only need the `Promise.resolve().then(...` if it's potentially going to throw.  In this case, you can just remove that added layer of wrapping.

Reply:
That is, this formulation:

```js
return new opts.Promise((resolve, reject) => {
  doSomething()
  thingThatReturnsPromise(...args).then(resolve, reject)
})
```

has two effects:

1. It guarantees that a promise _will_ be returned.
2. It guarantees that the _type_ of Promise will be the one in opts.Promise.

In our case, we can just do:

```js
doSomething()
return thingThatReturnsPromise(...args)
```

because we use native Promises everywhere.

--------

This formulation has two effects, also:

```js
return BB.try(() => {
  doSomethingThatMightThrow()
  return thingThatReturnsPromise()
})
```

1. If `doSomethingThatMightThrow()` throws, then the return value is a rejected Promise.;
2. The return value is converted to a bluebird Promise.

In that case, we can do:

```js
return Promise.resolve().then(() => {
  doSomethingThatMightThrow()
  return thingThatReturnsPromise()
})
```

because it will catch thrown errors (as they're thrown in a deferred Promise chain), and we use native Promises everywhere, so no need to convert anything.